File: fAiL.c
#include <time.h>
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

#define LEN 6

/* letters are all lowercase so I don't need to convert to lower later */
char alnum[] = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz0123456789";

int main()
{
	srand(time(NULL));

	char buf[LEN];

	for (int i = 0; i < 5000000; i++) {
		for (int j = 0; j < LEN; j++) {
			buf[j] = alnum[rand()%62];
		}

		if (strstr(buf, "fail")) {
			printf("hit: %s pos: %d\n", buf, i);
		}
	}
}

File: fAiL.go
package main

import (
    "fmt"
    "math/rand"
    "time"
    "strings"
)

var alnum = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"


func main() {
	rand.Seed(time.Now().UnixNano())

	for i := 0; i < 5000000; i++ {
		buf := make([]byte, 6)
		for j := 0; j < 6; j++ {
			buf[j] = alnum[rand.Intn(62)]
		}
		file_name := string(buf)

		if strings.Contains(strings.ToLower(file_name), "fail") {
			fmt.Printf("hit: %s pos: %d\n", file_name, i)
		}
	}
}

